### PR TITLE
Enable Travis CI

### PR DESCRIPTION
**Pull request by texasaggie97**
[X] This contribution adheres to [CONTRIBUTING.md]
### What does this Pull Request accomplish?
* Enable Travis CI integration on GitHub
* Add support for using tox for testing
* We do not use tox-travis integration
  * Travis "builds" with one version - 3.4
  * The build then invokes tox which iterates over the listed versions
* Adds 'make test' target
  * Copies tox.ini to bin/<driver> and then invokes tox at that location
* Make all still invokes pytest using the same python that is used to build

### Why should this Pull Request be merged?
* Automation is good

### What testing has been done?
* [Travis CI](https://travis-ci.org/ni/nimi-python)

### CODE REVIEW

File: build/Makefile [hunk 1]
```diff
@@ -22,15 +22,17 @@ endef
...
       $(eval $(call per_driver_per_target,$(d),$(t)))))
 
 define per_target
 $1:
 	$(foreach d,$(DRIVERS),make --no-print-directory -f src/$(d)/$(d).mak DRIVER=$(d) $1 &&)  echo
 endef
 
-$(foreach t,$(POSSIBLE_TARGETS),$(eval $(call per_target,$(t))))
+# We add test when we create the $(POSSIBLE_TARGETS). This is because it shouldn't be
+# run when all us used, but we want it to work
```
Comment (marcoskirsch) [inline build/Makefile]: us -> is

File: build/templates/attributes.py.mako [hunk 2]
```diff
@@ -1,17 +1,23 @@
 <%
```
Comment (marcoskirsch) [inline build/templates/attributes.py.mako]: We should remove this file, it's not being used. It should be its own PR.

**Comment by marcoskirsch** [comment]
There appears to be changes in the diff that are not mentioned in the PR description. Are we missing stuff?

File: build/Makefile (hunk 1 above)
Comment (texasaggie97) [inline build/Makefile]: Fixed

File: build/templates/attributes.py.mako (hunk 2 above)
Comment (texasaggie97) [inline build/templates/attributes.py.mako]: Done in #68
